ti/instrumentation/java/helper.go

```diff
@@ -108,18 +177,55 @@ func ParseJavaNode(filename string) (*Node, error) {
 		node.File = parts[len(parts)-1]
 		node.Lang = LangType_JAVA
 	} else if strings.HasSuffix(filename, ".scala") {
-		// Cannot make any assumption from scala/kotlin file path, return generic source node for now
-		node.Lang = LangType_JAVA
+		// If the scala filepath does not match any of the test paths below, return generic source node
 		node.Type = NodeType_SOURCE
+		node.Lang = LangType_JAVA
 		f := strings.TrimSuffix(filename, ".scala")
 		parts := strings.Split(f, "/")
 		node.Class = parts[len(parts)-1]
+		// Check for Test Node
+		if strings.Contains(filename, SCALA_TEST_PATH) {
+			r = regexp.MustCompile(`^.*src/test/scala/`)
```

Comment:
Why don't we replace ^.*src/test/scala/ with a constant? Same for java

Reply:
Done